core/src/plugins/filed/python/pyfiles/BareosFdPluginBaseclass.py

```diff
@@ -30,11 +30,86 @@
 import stat
 import sys
 import time
+from base64 import a85decode
+
+
+def parse_plugindef_string(plugindef):
+    options = {}
+    parts = plugindef.split(":")
+    while parts:
+        raw_key, _, raw_val = parts.pop(0).partition("=")
+        key = raw_key.strip()
+        val = raw_val.strip()
+        if not key:
+            continue
+        if not val:
+            bareosfd.JobMessage(
+                M_WARNING, "ignoring option '%s' with no value!\n" % (key)
+            )
+            continue
+        # See if the last character is a escape of the value string
+        while val[-1:] == "\\":
+            val = val[:-1] + ":" + plugin_options.pop(0)
+        if key not in options:
+            options[key] = val
+    return options
+
+
+def parse_options_file(path):
+    bareosfd.DebugMessage(100, 'reading options from file "%s"\n' % (path))
+    options = {}
+    with open(path, "r") as fp:
+        lines = fp.readlines()
+        while lines:
+            raw_key, _, raw_val = lines.pop(0).partition("=")
+            key = raw_key.strip()
+            val = raw_val.strip()
+            if key == "" and val == "":  # empty line
+                continue
+            if key.startswith(("#", ";", "[")):
+                continue
+            elif not val:
+                bareosfd.JobMessage(
+                    M_WARNING, "ignoring option '%s' with no value!\n" % (key)
+                )
+                continue
+            # check if last character escapes the newline
+            while val[-1:] == "\\":
+                val = val[:-1] + lines.pop(0).strip()
+            if key not in options:
+                options[key] = val
```

Comment:
In what case? We cannot catch the case where an option was set twice, because that will happen all the time when running a restore IIRC.

Reply:
One way to prevent this is to only set a flag when an override was detected (maybe also save information related to the overwrite; i.e. which option was overridden).  Then we can emit a warning if we catch `bEventStartBackupJob`.

Reply:
Here we can only see the options in the file we're currently parsing. So we can actually catch that.